p2p/net/gostream/listener.go

```diff
@@ -60,6 +63,7 @@ func Listen(h host.Host, tag protocol.ID) (net.Listener, error) {
 	}
 
 	h.SetStreamHandler(tag, func(s network.Stream) {
+		s.SetDeadline(time.Now().Add(ServerTimeout))
```

Comment:
This seems like the wrong place to set the deadline. The consumer might want to use the stream for a long time.

Reply:
You're right. This `Conn` this transforms to should have the deadline set by the user.